reporting/src/main/java/org/wso2/testgrid/reporting/ChartGenerator.java

```diff
@@ -0,0 +1,152 @@
+package org.wso2.testgrid.reporting;
+
+import javafx.application.Platform;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.embed.swing.JFXPanel;
+import javafx.embed.swing.SwingFXUtils;
+import javafx.scene.Scene;
+import javafx.scene.chart.CategoryAxis;
+import javafx.scene.chart.Chart;
+import javafx.scene.chart.NumberAxis;
+import javafx.scene.chart.PieChart;
+import javafx.scene.chart.StackedBarChart;
+import javafx.scene.chart.XYChart;
+import javafx.scene.image.WritableImage;
+import javafx.stage.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.reporting.model.email.BuildExecutionSummary;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Map;
+import javax.imageio.ImageIO;
+
+/**
+ * This class is responsible for generating the necessary charts for the email report.
+ *
+ * @since 1.0.0
+ */
+public class ChartGenerator {
+
+    private static final Logger logger = LoggerFactory.getLogger(ChartGenerator.class);
+    private static  final String summaryChartFileName = "summary.png";
+    private static final String historyChartFileName = "history.png";
+    private String chartGenLocation;
+
+    public ChartGenerator(String chartGenLocation) {
+        new JFXPanel();
+        this.chartGenLocation = chartGenLocation;
+    }
+
+    /**
+     * Generates a pie chart with the summary test results of the current build.
+     *
+     * @param passedCount  passed test count
+     * @param failedCount  failed test count
+     * @param skippedCount skipped test count
+     * @throws IOException if the chart cannot be written into a file
+     */
+    public void generateSummaryChart(int passedCount, int failedCount, int skippedCount) {
+
+        ObservableList<PieChart.Data> pieChartData =
+                FXCollections.observableArrayList(
+                        new PieChart.Data("Failed", failedCount),
+                        new PieChart.Data("Skipped", skippedCount),
+                        new PieChart.Data("Passed", passedCount));
+        final PieChart chart = new PieChart(pieChartData);
+        chart.setAnimated(false);
+        chart.setTitle("Build Failure Summary by Test Plans");
+        genChart(chart, 500, 500, summaryChartFileName);
```

Comment:
What are 500s here?

Reply:
This is the size of the generated graph.